tests/utils.rs

```diff
@@ -0,0 +1,271 @@
+            if !maps_match(&real_claims, &claims) {
+                panic!("IDENTITY: User claims differ")
+            }
+        }
+        _ => {
+            unreachable!(
+                "Unreachable metastate reply for the IdentityStateQuery::UserClaims payload has occured"
+            )
+        }
+    }
+}
+
+pub fn check_claim_state_utils(
+    id_program: &Program,
+    subject: PublicKey,
+    piece_id: PieceId,
+    claim: Claim,
+) {
+    match id_program.meta_state(IdentityStateQuery::Claim(subject, piece_id)) {
+        gstd::Ok(IdentityStateReply::Claim(real_claim)) => {
+            if let Some(cl) = real_claim {
+                if claim != cl {
+                    panic!("IDENTITY: Claims differ");
+                }
+            } else {
+                panic!("IDENTITY: No such claim");
+            }
+        }
+        _ => {
+            unreachable!(
+                "Unreachable metastate reply for the IdentityStateQuery::Claim payload has occured"
+            )
+        }
+    }
+}
+
+pub fn check_verifiers_state_utils(
+    id_program: &Program,
+    subject: PublicKey,
+    piece_id: PieceId,
+    verifiers: Vec<PublicKey>,
+) {
+    match id_program.meta_state(IdentityStateQuery::Verifiers(subject, piece_id)) {
+        gstd::Ok(IdentityStateReply::Verifiers(real_verifiers)) => {
+            if real_verifiers != verifiers {
+                panic!("IDENTITY: Verifiers differ");
+            }
+        }
+        _ => {
+            unreachable!(
+                "Unreachable metastate reply for the IdentityStateQuery::Verifiers payload has occured"
+            )
+        }
+    }
+}
+
+pub fn check_date_state_utils(
+    id_program: &Program,
+    subject: PublicKey,
+    piece_id: PieceId,
+    date: u128,
+) {
+    match id_program.meta_state(IdentityStateQuery::Date(subject, piece_id)) {
+        gstd::Ok(IdentityStateReply::Date(real_date)) => {
+            if real_date != date {
+                panic!("IDENTITY: Dates differ");
+            }
+        }
+        _ => {
+            unreachable!(
+                "Unreachable metastate reply for the IdentityStateQuery::Date payload has occured"
+            )
+        }
+    }
+}
+
+pub fn check_valid_state_utils(
+    id_program: &Program,
+    subject: PublicKey,
+    piece_id: PieceId,
+    valid: bool,
+) {
+    match id_program.meta_state(IdentityStateQuery::ValidationStatus(subject, piece_id)) {
+        gstd::Ok(IdentityStateReply::ValidationStatus(real_valid)) => {
+            if real_valid != valid {
+                panic!("IDENTITY: Validation status differ");
+            }
+        }
+        _ => {
+            unreachable!(
+                "Unreachable metastate reply for the IdentityStateQuery::ValidationStatus payload has occured"
+            )
+        }
+    }
+}
+
+fn maps_match<
+    T: Eq + gstd::Ord + gstd::Clone,
+    U: gstd::Clone + gstd::PartialEq<V>,
+    V: gstd::Clone,
```

Comment:
Aren't these traits in preamble?
```suggestion
    T: Eq + Ord + Clone,
    U: Clone + PartialEq<V>,
    V: Clone,
```